controllers/drpolicy_controller.go

```diff
@@ -0,0 +1,282 @@
+	MirroringModeKey                      = "mirroringMode"
+	SchedulingIntervalKey                 = "schedulingInterval"
+	ReplicationSecretNameKey              = "replication.storage.openshift.io/replication-secret-name"
+	ReplicationSecretNamespaceKey         = "replication.storage.openshift.io/replication-secret-namespace"
+	ReplicationIDKey                      = "replicationid"
+	RBDVolumeReplicationClassNameTemplate = "rbd-volumereplicationclass-%v"
+	RBDReplicationSecretName              = "rook-csi-rbd-provisioner"
+	RamenLabelTemplate                    = "ramendr.openshift.io/%s"
+	RBDProvisionerTemplate                = "%s.rbd.csi.ceph.com"
+)
+
+type DRPolicyReconciler struct {
+	HubClient client.Client
+	Scheme    *runtime.Scheme
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *DRPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	dpPredicate := utils.ComposePredicates(predicate.GenerationChangedPredicate{})
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&ramenv1alpha1.DRPolicy{}, builder.WithPredicates(dpPredicate)).
+		Complete(r)
+}
+
+func (r *DRPolicyReconciler) getMirrorPeerForClusterSet(ctx context.Context, clusterSet []string) (*multiclusterv1alpha1.MirrorPeer, error) {
+	var mpList multiclusterv1alpha1.MirrorPeerList
+	err := r.HubClient.List(ctx, &mpList)
+	if err != nil {
+		klog.Error("could not list mirrorpeers on hub")
+		return nil, err
+	}
+
+	if len(mpList.Items) == 0 {
+		klog.Info("no mirrorpeers found on hub yet")
+		return nil, fmt.Errorf("resource not found")
+	}
+	for _, mp := range mpList.Items {
+		if (mp.Spec.Items[0].ClusterName == clusterSet[0] && mp.Spec.Items[1].ClusterName == clusterSet[1]) ||
+			(mp.Spec.Items[1].ClusterName == clusterSet[0] && mp.Spec.Items[0].ClusterName == clusterSet[1]) {
+			klog.Infof("found mirrorpeer %q for drpolicy", mp.Name)
+			return &mp, nil
+		}
+	}
+
+	klog.Info("could not find any mirrorpeer for drpolicy")
+	return nil, nil
+}
+func (r *DRPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.Infof("running DRPolicy reconciler on hub cluster")
+	// Fetch DRPolicy for given Request
+	var drpolicy ramenv1alpha1.DRPolicy
+	err := r.HubClient.Get(ctx, req.NamespacedName, &drpolicy)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Info("Could not find DRPolicy. Ignoring since object must have been deleted")
+			return ctrl.Result{}, nil
+		}
+		klog.Error(err, "Failed to get DRPolicy")
+		return ctrl.Result{}, err
+	}
+
+	if !drpolicy.GetDeletionTimestamp().IsZero() {
+		result, err := r.deleteManifestWorks(ctx, &drpolicy)
+		if err != nil {
+			return result, err
+		}
+		err = r.HubClient.Get(ctx, req.NamespacedName, &drpolicy)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				klog.Info("Could not find DRPolicy. Ignoring since object must have been deleted")
+				return ctrl.Result{}, nil
+			}
+			klog.Error(err, "Failed to get DRPolicy")
+			return ctrl.Result{}, err
+		}
```

Comment:
This is not required. Why do you want to get the object that is being deleted or already deleted?